File: frontend/src/types/security.ts
/**
 * Zenith AI — Sovereign Industrial AI Workbench
 * Security, Sovereignty & Audit Domain Types
 */

export type DataSourceOrigin =
  | 'LIVE_API'
  | 'DEV_TEST_DATA'
  | 'UNAVAILABLE'

export interface SecurityMetric<T = number> {
  value: T | null
  origin: DataSourceOrigin
  note?: string
  lastUpdated?: string
}

export interface SovereigntySummary {
  externalApiCalls: SecurityMetric<number>
  cloudModelCalls: SecurityMetric<number>
  externalDnsRequests: SecurityMetric<number>
  networkEgressBytes: {
    bytes: number | null
    formatted: string
    origin: DataSourceOrigin
    note?: string
    lastUpdated?: string
  }
  localServiceAvailabilityPercent: SecurityMetric<number>
}

export type LocalServiceStatus =
  | 'HEALTHY'
  | 'DEGRADED'
  | 'UNAVAILABLE'
  | 'MAINTENANCE'

export interface LocalServiceHealthItem {
  id: string
  name: string
  status: LocalServiceStatus
  latencyMs: number | null
  uptime: string | null
  memoryVram: string | null
  version: string | null
  dataOrigin: DataSourceOrigin
}

export type NetworkRequestType =
  | 'LOCAL_IPC'
  | 'BLOCKED_EGRESS'
  | 'INTERNAL_RPC'

export type NetworkRequestStatus =
  | 'ALLOWED'
  | 'BLOCKED_BY_AIRGAP'
  | 'VERIFIED'

export interface NetworkMonitorEntry {
  id: string
  timestamp: string
  service: string
  requestType: NetworkRequestType
  targetEndpoint: string
  status: NetworkRequestStatus
  protocol: string
  dataOrigin: DataSourceOrigin
}

export type AuditStatus =
  | 'CONFIRMED'
  | 'BLOCKED'
  | 'FLAGGED'

export interface AuditTimelineEntry {
  id: string
  timestamp: string
  user: string
  action: string
  resource: string
  model: string
  tool: string
  status: AuditStatus
  checksumSha256?: string
  dataOrigin?: DataSourceOrigin
}

export type SecurityEventType =
  | 'WARNING'
  | 'BLOCKED_REQUEST'
  | 'AUTH_EVENT'
  | 'FILE_ACCESS'

export type SecuritySeverity =
  | 'HIGH'
  | 'MEDIUM'
  | 'LOW'
  | 'INFO'

export interface SecurityEvent {
  id: string
  type: SecurityEventType
  severity: SecuritySeverity
  title: string
  detail: string
  timestamp: string
  dataOrigin: DataSourceOrigin
}

File: backend/app/schemas/security.py
from typing import List, Optional
from pydantic import BaseModel


class SecurityMetricSchema(BaseModel):
    value: Optional[int] = None
    origin: str = "LIVE_API"
    note: Optional[str] = None
    last_updated: Optional[str] = None


class NetworkEgressSchema(BaseModel):
    bytes: Optional[int] = None
    formatted: str = "0 Bytes"
    origin: str = "LIVE_API"
    note: Optional[str] = None
    last_updated: Optional[str] = None


class SovereigntySummaryResponse(BaseModel):
    external_api_calls: SecurityMetricSchema
    cloud_model_calls: SecurityMetricSchema
    external_dns_requests: SecurityMetricSchema
    network_egress_bytes: NetworkEgressSchema
    local_service_availability_percent: SecurityMetricSchema


class LocalServiceHealthItemSchema(BaseModel):
    id: str
    name: str
    status: str
    latency_ms: Optional[int] = None
    uptime: Optional[str] = None
    memory_vram: Optional[str] = None
    version: Optional[str] = None
    data_origin: str = "LIVE_API"


class SecurityStatusResponse(BaseModel):
    sovereignty: SovereigntySummaryResponse
    services: List[LocalServiceHealthItemSchema]
